src/Shared/Components/KeyValueTable/KeyValueTable.component.tsx

```diff
@@ -0,0 +1,286 @@
+                                    style={{
+                                        ['--rotateBy' as string]: sortOrder === SortingOrder.ASC ? '0deg' : '180deg',
+                                    }}
+                                />
+                            </button>
+                        ) : (
+                            <div
+                                key={key}
+                                className={`cn-9 fs-13 lh-20 py-8 px-12 fw-6 flexbox dc__align-items-center dc__gap-2 ${key === firstHeaderKey ? 'head__key' : 'flex-grow-1'} ${className || ''}`}
+                            >
+                                {label}
+                            </div>
+                        ),
+                    )}
+                    {!!headerComponent && <div className="px-12">{headerComponent}</div>}
+                </div>
+                {!isAdditionNotAllowed && (
+                    <div
+                        className="table-row flexbox dc__align-items-center dc__border-bottom"
+                        style={{ borderColor: 'var(--N100)' }}
+                    >
+                        {headers.map(({ key }) => (
+                            <div
+                                key={key}
+                                className={`cn-9 fs-13 lh-20 py-8 px-12 flex dc__overflow-auto ${key === firstHeaderKey ? 'head__key' : 'flex-grow-1'}`}
+                            >
+                                <textarea
+                                    ref={key === firstHeaderKey ? inputRowRef : undefined}
+                                    className="table-input table-input__text-area pt-8 pb-8 pl-10 pb-10 lh-20 fs-13 fw-4"
+                                    value=""
+                                    rows={1}
+                                    placeholder={key === firstHeaderKey ? 'Enter Key' : 'Enter Value'}
+                                    onChange={onNewRowEdit(key)}
+                                />
+                            </div>
+                        ))}
+                    </div>
+                )}
+                {_rows?.map((row, index) => (
+                    <div
+                        key={`${index.toString()}`}
+                        className={`table-row flexbox dc__align-items-center ${index !== _rows.length - 1 ? 'dc__border-bottom' : ''}`}
+                        style={{ borderColor: 'var(--N100)' }}
+                    >
+                        {headers.map(({ key }, i) => (
+                            <div
+                                key={`${index.toString()}-${i.toString()}`}
+                                className={`cn-9 fs-13 lh-20 py-8 px-12 dc__overflow-auto flexbox dc__align-items-center dc__gap-4 ${key === firstHeaderKey ? 'head__key' : 'flex-grow-1'}`}
+                            >
+                                {maskValue?.[key] && row.data[key].value ? (
+                                    '*****'
```

Comment:
We can use DEFAULT_SECRET_PLACEHOLDER, have shipped that in common in this packet itself, so can add that while finishing